README.md

```diff
@@ -1,64 +1,129 @@
+creation of the following K8s resources:
+
+- A statefulSet whose pod spec has two containers, tha actual Jupyter server and
+  an [oauth2 proxy](https://github.com/oauth2-proxy/oauth2-proxy) which is
+  running in front of the Jupyter server
+- A PVC which will be mounted into the Jupyter server
+- A configmap to hold some non-secret configuration
+- A secret to hold some secret configuration
+- A service to expose the pod defined in the statefulSet
+- An ingress to make the jupyter server outside reachable from outside of the
+  cluster
+
+## Patching a JupyterServer
+
+We intentionally keep the configuration options through the jinja templates
+relatively limited to cover only what we believe to be the frequent use cases.
+However, as part of the custom resource spec, one can pass a list of
+[json](https://datatracker.ietf.org/doc/html/rfc6902) or
+[json merge](https://datatracker.ietf.org/doc/html/rfc7386) patches, which will
+be applied to the resource specifications _after_ the rendering of the Jinja
+templates. Through patching, one has the complete freedom to add, remove or
+change K8s resources which are created as part of the custom resource object.
 
 ## Motivation and use cases
 
 The main use case of Amalthea is to provide a layer on top of which developers
 can build kubernetes-native applications that allow their users to spin-up and
-manage Jupyter servers. We do not see Amalthea as a standalone tool which can be
-used by end users as creating Jupyter servers with Amalthea still requires
-access to the kubernetes API.
+manage Jupyter servers. We do not see Amalthea as a standalone tool which is
+used by end users, as creating Jupyter servers with Amalthea requires access to
+the Kubernetes API.
 
 ### Comparison to JupyterHub
 
 [JupyterHub](https://jupyterhub.readthedocs.io/en/stable/) is the standard
 application for serving Jupyter servers to multiple users. Unlike Amalthea,
 JupyterHub _is_ designed to be an application for the end user to interact with,
 and it can run on Kubernetes as well as on standalone servers. It therefore
-comes "batteries included" with a web frontend, user management, a database to
-keep track of running servers, a configurable web proxy, etc.
+comes "batteries included" with a web frontend, user management, a database that
+keeps track of running servers, a configurable web proxy, etc.
 
 The intended scope of Amalthea is much smaller than that. Specifically:
 
-- Amalthea requires that there is already some OpenID Connect provider in the
+- Amalthea requires that there is already an OpenID Connect provider in the
   application stack.
 - Amalthea itself is stateless. All state is stored as Kubernetes objects in
   etcd.
-- Amalthea uses the Kubernetes ingresses system for adding and removing
-  routes to Jupyter servers.
+- Amalthea uses the Kubernetes ingress- and service concepts for dynamically
+  adding and removing routes as Jupyter servers come and go.
```

Comment:
```suggestion
- Amalthea uses the Kubernetes-native ingress- and service concepts for dynamically
  adding and removing routes as Jupyter servers come and go, instead of relying on 
  an additoinal proxy for routing.
```